patterns/1-initial/extensions-to-manage-contributions-at-scale.md

```diff
@@ -0,0 +1,81 @@
+
+In order for the extensions model to be successful, there are few architectural considerations to keep in mind:
+
+1. <b>Easy to create:</b> To obtain community participation, extensions need to be easy to create.
+   - Create Github templates for extensions to allow addition of new features in new repositories, retaining the same modular structure as the primary repository, with the framework to package and release extensions.
+      - Ensure that as the primary repository changes, the template(s) are well-maintained.
+   - Add example extension(s) developed from the template, which project developers can reference to understand how to write a well-patterned extension.
+   - Loosen the requirements for contributors to create extensions by bypassing reviews to allow for faster release or experimentation.
+2. <b>Loose coupling:</b> Having modular components that contain functionality can allow loose coupling, where changes to extensions do not impact the quality of the main codebase or other extensions.
+3. <b>Dependency management:</b>  Each extension should be careful to pin the version range of the primary repository that it is built against (the same way it would any other dependency) and should be careful in its use of other dependencies that shadow dependencies of the primary repository such that the versions it chooses for those dependencies are compatible with the primary repository versions selected. Any conflicts with primary repository will be caught in the test framework for the extension.
+4. <b>Testing strategy:</b> How to test extensions both individually and in combination?
+   - <b>Testing extension individually:</b> Extensions template will provide a test framework to be used by the extension developers to test the capability added. This can include a framework for unit tests, runtime performance and quality tests.
+   - <b>Testing extension in combination with primary repository:</b> Extension developers have a well-patterned method for testing their extension against specific versions of the primary repository without involvement from the primary repository's maintainers.
+   - <b>Testing extension in combination with other extensions:</b> Providing a test framework for this scenario could result in being excessive especially if there are a large number of extensions that are still being explored by users and unlikely to be all used in combination. If a user runs into conflicts while using extensions in combination (which should be unlikely with sufficient loose coupling), the user can raise an issue to the respective extension owners who will sort it out. As an extension reaches later phases of the lifecycle and gets merged into the primary repository, it would be tested in combination with rest of library and any dependency conflicts will have to be resolved at the time.
```

Comment:
This is where you start calling the core project a "library".
I wonder if this is essential?

I would suggest to pick one terminology at the very beginning to refer to the "core project", and then stick to that term for the rest of the pattern.

Reply:
removed reference to library